chia/consensus/block_creation.py

```diff
@@ -213,8 +214,17 @@ def create_foliage(
         additions_root = addition_merkle_set.get_root()
         removals_root = removal_merkle_set.get_root()
 
-        generator_hash = solution_program.get_tree_hash() if solution_program is not None else bytes32([0] * 32)
+        generator_hash = bytes32([0] * 32)
+        if solution_program is not None:
+            generator_hash = std_hash(solution_program)
+
         generator_refs_hash = bytes32([1] * 32)
+        if generator_block_heights_list not in (None, []):
+            generator_ref_list_bytes = b"".join(
+                [(i).to_bytes(4, byteorder="big") for i in generator_block_heights_list]
```

Comment:
At a block every 20 seconds, we're going to exceed 32-bit block numbers in under eight years. This seems bad.

Reply:
If we make these relative values, then I think we're okay (I bet we could get away with two bytes in that case).

Reply:
:) I have notes in my first draft about this very issue (that's the type for block height on the wire, too) - I talked it over with Straya and Mariano, and they convinced me that 32bit block height is okay.

Reply:
Okay, got it.